### ISSUE TITLE
Consider foreign key constraints

### ISSUE THREAD

**Issue by qsona** [mbleigh/seed-fu#101]
If `table_a` has `table_b_id` column, and the foreign key constraint exists, `seed_fu` try to insert initial `table_a` data and fails because of the constraint.

As far as I know, seed_fu reads files as alphabetical order of filenames, so I can change the filenames as
`db/fixtures/01_table_b.rb` and `db/fixtures/02_table_a.rb` to fix it, but I think it's better if we need not to do it.
Is there any solutions? Thanks.

**Comment by abrthel** [comment]
This was an issue for me as well. To get around it I created another rake task like so

```Ruby
  # taken and modified from: https://github.com/mbleigh/seed-fu/blob/master/lib/tasks/seed_fu.rake
  task seed_fu: :environment do
    if ENV["FILTER"]
      filter = /#{ENV["FILTER"].gsub(/,/, "|")}/
    end

    if ENV["FIXTURE_PATH"]
      fixture_paths = [ENV["FIXTURE_PATH"], ENV["FIXTURE_PATH"] + '/' + Rails.env]
    else
      fixture_paths = ['db/fixtures/**/*']
    end

    connection = ActiveRecord::Base.connection
    connection.disable_referential_integrity do
      SeedFu.seed(fixture_paths, filter)
    end
  end
```

This is probably a terrible idea, but I just needed a way to load fixture files from folders at varying depths and I didn't have a reliable way to preorder my fixtures in advance.

**Comment by stuarthannig** [comment]
Yes,foreign key constraint consideration would be nice. Working around it myself.

**Comment by eliotsykes** [comment]
Disabling referential integrity during seeding is a workaround for some situations, but note postgresql (and I'm assuming other dbs) will not validate the constraints afterward, so you can end up with orphaned foreign keys:

```rb
ActiveRecord::Base.connection.disable_referential_integrity do
  SeedFu.seed
end
```
